app/redux/shared_actions/DeviceActionCreators.js

```diff
@@ -199,6 +200,17 @@ export const reconnectingToDevice = (name) => {
 
 export const receivedLiftData = (isValid, data, time=new Date()) => (dispatch, getState) => {
     const state = getState();
+    const workingSet = SetsSelectors.getWorkingSet(state);
+    const workingSetReps = workingSet.reps
+    const lastRep = workingSetReps[workingSetReps.length - 1];
+    const currentRep = RepDataMap.averageVelocity(data);
+    const lastRepVelocity = lastRep ? RepDataMap.averageVelocity(lastRep.data) : null;
+
+    if (lastRepVelocity && (lastRepVelocity * .7) > currentRep) {
+        console.tron.log(currentRep + " dropped from " + lastRepVelocity);
+        dispatch(VelocityDropActionCreators.velocityDropped())
+    }
+
```

Comment:
For simplicity, we want to play a sound when the average velocity drops below the predetermined threshold. So if the user sets the velocity threshold to 0.6 in Settings, any rep that falls below 0.6 will cause a sound to play.

Full on velocity drop where the user sets the % in Settings and calculation is done via fastest rep vs current rep may be added at a later date if there's enough demand for it. I argue that it's slightly more complex as you need to store values and compare them, but not overly so. The bigger issue is settings.

So given that, I don't think you need to add anything to DeviceActionCreators. If anything, you just need a saga to listen to ADD_REP_DATA.